client/account-fees/index.js

```diff
@@ -0,0 +1,148 @@
+import { dateI18n } from '@wordpress/date';
+import moment from 'moment';
+
+/**
+ * Internal dependencies
+ */
+import ProgressBar from 'components/progress-bar';
+
+const currency = new Currency();
+
+const ExpirationBar = ( {
+	feeData: {
+		volume_allowance: volumeAllowance,
+		current_volume: currentVolume,
+	},
+} ) => {
+	if ( ! volumeAllowance ) {
+		return null;
+	}
+	return (
+		<ProgressBar
+			progressLabel={ currency.formatCurrency( currentVolume / 100 ) }
+			totalLabel={ currency.formatCurrency( volumeAllowance / 100 ) }
+			progress={ currentVolume / volumeAllowance }
+		/>
+	);
+};
+
+const ExpirationDescription = ( {
+	feeData: { volume_allowance: volumeAllowance, end_time: endTime },
+} ) => {
+	let description;
+	if ( volumeAllowance && endTime ) {
+		description = sprintf(
+			/* translators: %1: total payment volume until this promotion expires %2: End date of the promotion */
+			__(
+				'Discounted base fee expires after the first %1$s of total payment volume or on %2$s.',
+				'woocommerce-payments'
+			),
+			currency.formatCurrency( volumeAllowance / 100 ),
+			dateI18n( 'F j, Y', moment( endTime ) )
+		);
+	} else if ( volumeAllowance ) {
+		description = sprintf(
+			/* translators: %1: total payment volume until this promotion expires */
+			__(
+				'Discounted base fee expires after the first %1$s of total payment volume.',
+				'woocommerce-payments'
+			),
+			currency.formatCurrency( volumeAllowance / 100 )
+		);
+	} else if ( endTime ) {
+		description = sprintf(
+			/* translators: %1: End date of the promotion */
+			__(
+				'Discounted base fee expires on %1$s.',
+				'woocommerce-payments'
+			),
+			dateI18n( 'F j, Y', moment( endTime ) )
+		);
+	} else {
+		return null;
+	}
+	return <p className="description">{ description }</p>;
+};
+
+const AccountFees = ( { accountFees } ) => {
+	const baseFee = accountFees.base;
+	let currentFee = baseFee;
+
+	let feeDescription = sprintf(
+		/* translators: %1: Percentage part of the fee. %2: Fixed part of the fee */
+		__( '%1$.1f%% + %2$s per transaction', 'woocommerce-payments' ),
+		currentFee.percentage_rate * 100,
+		currency.formatCurrency( currentFee.fixed_rate / 100 )
+	);
+
+	if ( accountFees.discount.length ) {
+		// TODO: Figure out how the UI should work if there are several "discount" fees stacked.
+		currentFee = accountFees.discount[ 0 ];
+		let percentage, fixed;
+
+		if ( currentFee.discount ) {
+			// Proper discount fee (XX% off)
+			percentage = baseFee.percentage_rate * ( 1 - currentFee.discount );
+			fixed = baseFee.fixed_rate * ( 1 - currentFee.discount );
+		} else {
+			// Custom base fee (2% + $.20)
+			percentage = currentFee.percentage_rate;
+			fixed = currentFee.fixed_rate;
+		}
+
+		let descriptionString = sprintf(
+			/* translators: %1 Base fee (that don't apply to this account at this moment), %2 and %3: Current fee (e.g: 2.9% + $.30) */
+			__(
+				'<s>%1$s</s> %2$.1f%% + %3$s per transaction',
```

Comment:
It's possible that this rounding could overstate the effective percentage fee by a small amount. At the moment this will only ever be in the merchant's favour, so I think we can go ahead. We might like to open an issue to investigate the best way to present the fee + discount information though.

Reply:
Yeah, you're right. In the design thread it showed `1.5% + $0.15`, so I interpreted that we are fine sacrificing precision for readability. We could show 2 decimals instead of 1, but there's always a chance we'll have to round up/down a bit even in that case 🤷